scss/_utilities.scss

```diff
@@ -87,3 +87,41 @@
 @include text-emphasis-variant('.text-warning', $brand-warning);
 
 @include text-emphasis-variant('.text-danger', $brand-danger);
+
+// Weight
+
+.text-black {
+  font-weight: 900;
+}
+
+.text-extra-bold {
+  font-weight: 800;
+}
+
+.text-bold {
+  font-weight: 700;
+}
+
+.text-semi-bold {
+  font-weight: 600;
+}
+
+.text-medium {
+  font-weight: 500;
+}
+
+.text-normal {
```

Comment:
This name is way too generic. "normal" doesn't suggest any relation to boldness/`font-weight`.